src/metrics/pipeline/applied/type.go

```diff
@@ -187,11 +187,18 @@ func (p Pipeline) At(i int) OpUnion { return p.operations[i] }
 
 // Equal determines whether two pipelines are equal.
 func (p Pipeline) Equal(other Pipeline) bool {
+	// keep in sync with OpUnion.Equal as go is terrible at inlining anything with a loop
 	if len(p.operations) != len(other.operations) {
 		return false
 	}
+
 	for i := 0; i < len(p.operations); i++ {
-		if !p.operations[i].Equal(other.operations[i]) {
+		if p.operations[i].Type != other.operations[i].Type {
+			return false
+		}
+
+		if p.operations[i].Type == pipeline.RollupOpType &&
+			!p.operations[i].Rollup.Equal(other.operations[i].Rollup) {
```

Comment:
do we need to worry about the transform case?

Reply:
yes, my bad, missed it